### PR TITLE
Fix agent executor predict

### PR DESCRIPTION
**Pull request by serena-ruan**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/serena-ruan/mlflow/pull/12286?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/12286/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 12286
```

</p>
</details>

### Related Issues/PRs

 #xxx

### What changes are proposed in this pull request?
Fix input data parsing for AgentExecutors. We don't convert it to ChatRequest format as some agents might not support those types. 

### How is this PR tested?

- [ ] Existing unit/integration tests
- [x] New unit/integration tests
- [ ] Manual tests

### Does this PR require documentation update?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

#### Should this PR be included in the next patch release?

`Yes` should be selected for bug fixes, documentation updates, and other small changes. `No` should be selected for new features and larger changes. If you're unsure about the release classification of this PR, leave this unchecked to let the maintainers decide.

<details>
<summary>What is a minor/patch release?</summary>

- Minor release: a release that increments the second part of the version number (e.g., 1.2.0 -> 1.3.0).
  Bug fixes, doc updates and new features usually go into minor releases.
- Patch release: a release that increments the third part of the version number (e.g., 1.2.0 -> 1.2.1).
  Bug fixes and doc updates usually go into patch releases.

</details>

- [ ] Yes (this PR will be cherry-picked and included in the next patch release)
- [x] No (this PR will be included in the next minor release)

### CODE REVIEW

File: mlflow/langchain/api_request_parallel_processor.py [hunk 1]
```diff
@@ -344,6 +345,11 @@ def _transform_request_json_for_chat_if_necessary(request_json, lc_model):
                 2. A boolean indicating whether or not the request was transformed from the OpenAI
                 chat format.
         """
+        # Avoid converting the request to LangChain's Message format if the chain
+        # is an AgentExecutor, as LangChainChatMessage might not be accepted by the chain
```
Comment (BenWilson2) [inline mlflow/langchain/api_request_parallel_processor.py:349]: Is this a fix that needs to happen in LangChain to support this format?

File: mlflow/langchain/api_request_parallel_processor.py [hunk 2]
```diff
@@ -344,6 +345,11 @@ def _transform_request_json_for_chat_if_necessary(request_json, lc_model):
                 2. A boolean indicating whether or not the request was transformed from the OpenAI
                 chat format.
         """
+        # Avoid converting the request to LangChain's Message format if the chain
```
Comment (B-Step62) [inline mlflow/langchain/api_request_parallel_processor.py:348]: Can we add docstring about why we do this transformation and when it is `necessary`?  It's super hard to see what kind of input cases are expected and when the exception (like agent) applies....

File: mlflow/langchain/api_request_parallel_processor.py [hunk 3]
```diff
@@ -344,6 +345,11 @@ def _transform_request_json_for_chat_if_necessary(request_json, lc_model):
...
+        # is an AgentExecutor, as LangChainChatMessage might not be accepted by the chain
+        if isinstance(lc_model, AgentExecutor):
+            return request_json, False
+
```
Comment (B-Step62) [inline mlflow/langchain/api_request_parallel_processor.py:352]: (Just filing a follow-up JIRA in fine) Shall we extract these input conversion things out of `APIRequest` to a separate utility module and add unit tests for them? This class seem to do too much more than it was supposed to do as a dataclass of API request object, resulting in low test coverage. I think it makes our life easier when we start to receive more change requests for RAG/agent handling post DAIS🤔

File: mlflow/langchain/utils.py [hunk 4]
```diff
@@ -314,8 +314,14 @@ def _validate_and_prepare_lc_model_or_path(lc_model, loader_fn, temp_dir=None):
             type(lc_model.llm).__name__,
         )
 
-    if isinstance(lc_model, langchain.agents.agent.AgentExecutor) and not any(
-        isinstance(lc_model.agent.llm_chain.llm, supported_llm) for supported_llm in _SUPPORTED_LLMS
+    if (
```
Comment (B-Step62) [inline mlflow/langchain/utils.py:317]: nit: Can we extract this condition as a self-explanatory function like `is_agent_contain_unsupported_llm(chain)`?

File: mlflow/langchain/api_request_parallel_processor.py (hunk 3 above)
Comment (serena-ruan) [reply mlflow/langchain/api_request_parallel_processor.py:352]: Created JIRA! Will work on it later

File: mlflow/langchain/api_request_parallel_processor.py (hunk 2 above)
Comment (serena-ruan) [reply mlflow/langchain/api_request_parallel_processor.py:348]: I think it's necessary for `ChatModel` to convert the input from dictionary format to ChatMessages, but for other model who originally accepts "messages" dict input it's not necessary.

File: mlflow/langchain/utils.py (hunk 4 above)
Comment (serena-ruan) [reply mlflow/langchain/utils.py:317]: Emm it's only used here and we'll need to extract _SUPPORTED_LLMS if moving out, maybe it's fine keeping here and add a comment?

Comment (serena-ruan) [reply mlflow/langchain/utils.py]: Updated!

**Review by B-Step62 — APPROVED** [review]
LGTM!!😄

**Pull request merged**